### PR TITLE
worker: Improve logging and test

### PR DESCRIPTION
**Pull request by Martchus**
* PR for debugging https://progress.opensuse.org/issues/90152
* See particular commit messages for details

### CODE REVIEW

**Review by okurz — APPROVED** [review]

File: lib/OpenQA/Worker/Job.pm [hunk 1]
```diff
@@ -479,16 +479,16 @@ sub _stop_step_5_2_upload {
 
     # do final status upload for selected reasons
     if ($reason eq WORKER_COMMAND_OBSOLETE) {
-        log_debug("Setting job $job_id to incomplete (obsolete)");
+        log_debug("Considering job $job_id as incomplete due to obsoletion");
```
Comment (kalikiana) [inline lib/OpenQA/Worker/Job.pm:482]: This doesn't seem to be covered by the unit tests? New or existing

Comment (Martchus) [reply lib/OpenQA/Worker/Job.pm:482]: I've seen the diff in codecov. Considering how trivial the uncovered changes are I thought it wasn't worth adding tests.

Comment (okurz) [reply lib/OpenQA/Worker/Job.pm:482]: I have 100% statement coverage as a goal so ... it's worth it :)

Comment (okurz) [reply lib/OpenQA/Worker/Job.pm:482]: Of course there are always alternatives, e.g. make it a single-line statement or just get rid of code :)

Comment (kalikiana) [reply lib/OpenQA/Worker/Job.pm:482]: > I've seen the diff in codecov. Considering how trivial the uncovered changes are I thought it wasn't worth adding tests.

If none of the code paths is covered, I find it concerning. It's not just the log message. That just makes it more obvious.

Comment (Martchus) [reply lib/OpenQA/Worker/Job.pm:482]: > It's not just the log message.

I know, hence my phrasing "trivial … changes".

**Review by kalikiana — APPROVED** [review]

**Pull request merged**